File: tools/ask_user.py
import os


def ask_user(question: str) -> str:
    """Ask the user a question and get their text response.
    Use this when you need clarification, confirmation, or additional information.

    Args:
        question: The question to ask the user
    """
    auto = os.environ.get("TINY_CODE_ASK_USER")
    if auto:
        print(f"\n--- AI asks: {question} ---")
        print(f"> [auto-answer: {auto}]")
        return f"User's response: {auto}"
    try:
        print(f"\n--- AI asks: {question} ---")
        print("> ", end="", flush=True)
        answer = input().strip()
        return f"User's response: {answer}"
    except (EOFError, KeyboardInterrupt):
        return "User cancelled the input"

File: internal/tools/ask_user.go
package tools

import "os"

// AskUser asks the user a question and returns their response, honoring the
// TINY_CODE_ASK_USER / TINYCODE_ASK_USER auto-answer for tests.
func AskUser(question string, ask func(string) string) string {
	if auto := os.Getenv("TINY_CODE_ASK_USER"); auto != "" {
		return "User's response: " + auto
	}
	if auto := os.Getenv("TINYCODE_ASK_USER"); auto != "" {
		return "User's response: " + auto
	}
	return "User's response: " + ask(question)
}
